ignite/metrics/mean_absolute_error.py

```diff
@@ -17,18 +17,18 @@ class MeanAbsoluteError(Metric):
 
     @reinit__is_reduced
     def reset(self) -> None:
-        self._sum_of_absolute_errors = 0.0
+        self._sum_of_absolute_errors = torch.tensor(0.0, device=self._device)
         self._num_examples = 0
 
     @reinit__is_reduced
     def update(self, output: Sequence[torch.Tensor]) -> None:
         y_pred, y = output
         absolute_errors = torch.abs(y_pred - y.view_as(y_pred))
-        self._sum_of_absolute_errors += torch.sum(absolute_errors).item()
+        self._sum_of_absolute_errors += torch.sum(absolute_errors).detach().to(self._device)
```

Comment:
Does detach earlier avoid some storage in graph ? I mean detach `absolute_errors` rather than the sum.

Reply:
Yes, good point. Detaching earlier we can avoid adding a node for backward op.

Reply:
Fixed this in 6edd30d.